webdriver/tests/execute_async_script/pinning.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+
+from webdriver.transport import Response
+
+from tests.support.asserts import assert_error, assert_success
+
+
+def pin_script(session, name, script):
+    body = {"script": script}
+
+    return session.transport.send(
+        "POST", "/session/{session_id}/execute/pin/{script_name}".format(
+            session_id=session.session_id,
+            script_name=name),
+        body)
+
+
+def execute_async_script(session, script, name=None, args=None):
+    if args is None:
+        args = []
+    body = {"script": script, "args": args}
+
+    return session.transport.send(
+        "POST", "/session/{session_id}/execute/async/{script_name}".format(
+            session_id=session.session_id,
+            script_name=name),
+        body)
+
+
+def test_execute_pinned(session):
+    pin_script(session, "One", "return 1;")
```

Comment:
You cannot return from an async script.

This applies generally throughout this file.